join-chat-to-teams-meeting/Chat Experience/client.js

```diff
@@ -8,127 +8,132 @@ let callAgent;
-const callStateElement = document.getElementById('call-state');
+const callStateElement = document.getElementById("call-state");
 
 const messagesContainer = document.getElementById("messages-container");
 const chatBox = document.getElementById("chat-box");
 const sendMessageButton = document.getElementById("send-message");
-const messagebox = document.getElementById("message-box");
+const messageBox = document.getElementById("message-box");
 
-var userId = '';
-var messages = '';
+var userId = "";
+var messages = "";
 
 async function init() {
-	const connectionString = "<SECRET_CONNECTION_STRING>";
-	const endpointUrl = "<ENDPOINT_URL>";
+  const connectionString = "<SECRET_CONNECTION_STRING>";
+  const endpointUrl = "<ENDPOINT_URL>";
+  const identityClient = new CommunicationIdentityClient(connectionString);
 
-	const identityClient = new CommunicationIdentityClient(connectionString);
+  let identityResponse = await identityClient.createUser();
+  userId = identityResponse.communicationUserId;
+  console.log(`\nCreated an identity with ID: ${identityResponse.communicationUserId}`);
 
-	let identityResponse = await identityClient.createUser();
-	userId = identityResponse.communicationUserId;
-	console.log(`\nCreated an identity with ID: ${identityResponse.communicationUserId}`);
+  let tokenResponse = await identityClient.getToken(identityResponse, ["voip", "chat"]);
 
-	let tokenResponse = await identityClient.getToken(identityResponse, [
-		"voip",
-		"chat",
-	]);
+  const { token, expiresOn } = tokenResponse;
+  console.log(`\nIssued an access token that expires at: ${expiresOn}`);
+  console.log(token);
 
-	const { token, expiresOn } = tokenResponse;
-	console.log(`\nIssued an access token that expires at: ${expiresOn}`);
-	console.log(token);
+  const callClient = new CallClient();
+  const tokenCredential = new AzureCommunicationTokenCredential(token);
+  callAgent = await callClient.createCallAgent(tokenCredential);
+  callButton.disabled = false;
 
-	const callClient = new CallClient();
-	const tokenCredential = new AzureCommunicationTokenCredential(token);
-	callAgent = await callClient.createCallAgent(tokenCredential);
-	callButton.disabled = false;
+  chatClient = new ChatClient(endpointUrl, new AzureCommunicationTokenCredential(token));
 
-	chatClient = new ChatClient(
-		endpointUrl,
-		new AzureCommunicationTokenCredential(token)
-	);
-
-	console.log('Azure Communication Chat client created!');
+  console.log("Azure Communication Chat client created!");
 }
 
 init();
 
+meetingLinkInput.addEventListener("input", () => {
+  // Chat thread Id is included in the meeting link, usually the
+  // part containing the word 'thread'
+  const url = new URL(meetingLinkInput.value);
+  const decodedPath = decodeURIComponent(url.pathname);
+  const threadMatcher = /(.*meetup-join\/)(?<threadId>19.*)(\/.*)/g;
+
+  for (const match of decodedPath.matchAll(threadMatcher)) {
+    threadIdInput.value = match.groups.threadId;
```

Comment:
It looks like `threadIdInput.value` will be set to the last value in `decodedPath.matchAll(threadMatcher)`. Should we just use the last one?